go/pkg/cache/singleflightcache/singleflightcache.go

```diff
@@ -35,12 +38,23 @@ func (c *Cache) LoadOrStore(key interface{}, valFn func() (interface{}, error))
 // Store accepts a value instead of a valFn since it is intended to be only used in
 // cases where updates are lightweight and do not involve computing the cache value.
 func (c *Cache) Store(key interface{}, val interface{}) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 	e := &entry{val: val}
 	e.compute.Do(func() {}) // mark as computed
 	c.store.Store(key, e)
 }
 
 // Delete removes a key from the cache.
 func (c *Cache) Delete(key interface{}) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 	c.store.Delete(key)
 }
+
+// Reset invalidates all cache entries.
+func (c *Cache) Reset() {
```

Comment:
@ramymedhat do we need this Reset somewhere? I couldn't find non-test use cases that we could probably tweak in a simpler manner.

Reply:
Happy to remove in another CL